### ISSUE TITLE
--mismatch option does not accept negative values

### ISSUE THREAD

**Issue by thiesgehrmann** [biocore/sortmerna#322]
Hi, I think there is a bug in the input processing of the mismatch argument.
SortMeRNA version 4.3.4

It seems that the mismatch argument is recognized as a boolean flag, rather than a parameter value?
It then recognizes the `-3` as a seperate argument:

```
> sortmerna \
    -ref reffile.fasta \
    -reads "readfile.fq" \
    --mismatch -3
[process:1376] === Options processing starts ... ===
Found value: sortmerna
Found flag: -ref
Found value: reffile.fasta of previous flag: -ref
Found flag: -reads
Found value: readfile.fq of previous flag: -reads
Found flag: --mismatch
Previous flag: --mismatch is Boolean. Setting to True
Found flag: -3
[process:1460] Processing option: 3 with value: 
[opt_default:943] ERROR: Option: '3' is not recognized
```

Providing a positive value gives the following:
```
[process:1460] Processing option: mismatch with value: 3

[opt_mismatch:385] ERROR: --mismatch [INT] takes a negative integer (ex. --mismatch -2)
```

Since the argument should be explicitly negative anyway, perhaps a simple fix would be to allow it to be positive, and then explicitly negate it before calculating the scoring matrix? i.e. `mismatch = -atoi(val.data());` @ https://github.com/biocore/sortmerna/blob/master/src/sortmerna/options.cpp#L382

Cheers,
-Thies

**Comment by biocodz** [comment]
Thank you for reporting this. Definitely needs a fix.

**Comment by biocodz** [comment]
Fixed in commit `deb68535b190b49ecabf2be74012ba3c36ef79be`. To appear in release `4.3.5` (soon)

**Comment by biocodz** [comment]
fixed in [v4.3.6](https://github.com/biocore/sortmerna/releases/tag/v4.3.6)
